gcloud-java-storage/src/main/java/com/google/gcloud/storage/Storage.java

```diff
@@ -755,12 +829,23 @@ public static Builder builder() {
   BlobInfo compose(ComposeRequest composeRequest);
 
   /**
-   * Send a copy request.
+   * Sends a copy request. Returns a {@link CopyWriter} object for the provided
+   * {@code CopyRequest}. If source and destination objects share the same location and storage
+   * class the source blob is copied with one request and {@link CopyWriter#result()} immediately
+   * returns, regardless of the {@link CopyRequest#megabytesCopiedPerChunk} parameter.
+   * If source and destination have different location or storage class {@link CopyWriter#result()}
+   * might issue multiple RPC calls depending on blob's size.
+   * <p>
+   * Example usage of copy:
+   * <pre>    {@code BlobInfo blob = service.copy(copyRequest).result();}
```

Comment:
I think this is great but I also think we should say a bit more about the option of using `isDone`
and `copyChunk` (could be here or in the `CopyWriter` class or maybe my suggestion for the `CopyWriter.response` javadoc suggestion is be sufficient?).